File: src/main/java/app/application/data/entities/YoutubePlaylistVideoDetail.java
package app.application.data.entities;

import com.github.kiulian.downloader.model.playlist.PlaylistVideoDetails;
import javafx.beans.property.BooleanProperty;
import javafx.beans.property.SimpleBooleanProperty;
import lombok.Getter;

public class YoutubePlaylistVideoDetail extends YoutubeEntity<PlaylistVideoDetails>{

	@Getter
	private final BooleanProperty ignore;

	public YoutubePlaylistVideoDetail(PlaylistVideoDetails playlistVideoDetails) {
		this.reference = playlistVideoDetails;
		this.ignore = new SimpleBooleanProperty(false);
	}

	public String getVideoThumbnailUrl(){
		return reference.thumbnails().get(0).split("\\?sqp")[0];
	}

	public String getVideoTitle(){
		return reference.title();
	}

	public String getVideoId(){
		return reference.videoId();
	}

}

File: src/main/kotlin/app/application/data/entities/YoutubePlaylistVideoDetail.kt
package app.application.data.entities

import com.github.kiulian.downloader.model.playlist.PlaylistVideoDetails
import javafx.beans.property.BooleanProperty
import javafx.beans.property.SimpleBooleanProperty
import javafx.scene.image.Image

class YoutubePlaylistVideoDetail(playlistVideoDetails: PlaylistVideoDetails?) : YoutubeEntity<PlaylistVideoDetails?>() {
    val ignore: BooleanProperty

    init {
        reference = playlistVideoDetails
        ignore = SimpleBooleanProperty(false)
    }

    private var _thumbnailImage: Image? = null

    val thumbnailImage: Image
    get() {
        if(_thumbnailImage == null){
            _thumbnailImage = Image(videoThumbnailUrl)
        }
        return _thumbnailImage as Image
    }

    private val videoThumbnailUrl: String?
        get() = reference?.thumbnails()?.get(0)?.split("\\?sqp")?.toTypedArray()?.get(0)

    val videoTitle: String?
        get() = reference?.title()

    val videoId: String?
        get() = reference?.videoId()
}